src/insexpand.c

```diff
@@ -5964,6 +5982,8 @@ ins_complete(int c, int enable_pum)
     else if (insert_match && stop_arrow() == FAIL)
 	return FAIL;
 
+    compl_curr_win = curwin;
+    compl_curr_buf = curwin->w_buffer;
```

Comment:
This seems not the only place where completion is started. There are other places like `set_completion()` too.

Reply:
but `set_completion` also calling `ins_complete`

Reply:
Oh, right